### PR TITLE
Embed: Add Wolfram Cloud as an embed provider

### PR DESCRIPTION
**Pull request by pento**
## Description

Following on from Wolfram Cloud being added as a trusted embed provider to WordPress Core ([WP53326](https://core.trac.wordpress.org/ticket/53326)), this PR adds Wolfram Cloud as an embed block provider.

The Wolfram icon comes from their website, it's a simplified version of their [main icon](https://company.wolfram.com/press-center/).

## How has this been tested?

- Tested against WordPress Master.
- Tested by pasting a variety of Wolfram Cloud URLs into the editor, and into the Wolfram Cloud embed block.

## Screenshots

Screenshot of Wolfram Cloud in the block selector
Screenshot of an embedded notebook in the editor

## Types of changes

New feature.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows the accessibility standards.
- [x] I've tested my changes with keyboard and screen readers.
- [x] My code has proper inline documentation.
- [x] I've included developer documentation if appropriate.
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by ntsekouras** [comment]
Thanks for the PR @pento ! A provided example link would be great for testing purposes - I created an account there to test and I'm still not sure if I published a 'right' thing there 😄 

Since the provider is to be included in `5.9`, does it make sense to include in the plugin now?

Also I'm wondering how this should be handled best as the plugin supports two WP versions behind.. Anyone knows how similar cases were handled? Maybe for `TikTok` that seems to be [the last one added](https://wordpress.org/support/article/embeds/)..

--cc @gziolo

**Comment by pento** [comment]
Thanks for testing, @ntsekouras. The [Wolfram Physics Project](https://www.wolframphysics.org/technical-introduction/) is a good source for test pages: pretty much any page will have a "Open in Wolfram Cloud" link at the bottom.

With the WordPress 5.9 feature freeze coming up, this needs to be merged ASAP. I don't recall seeing problems in the past of people trying to use embeds in the plugin before the appropriate Core version is released.

File: packages/block-library/src/embed/variations.js [hunk 1]
```diff
@@ -341,6 +342,14 @@ const variations = [
 		],
 		attributes: { providerNameSlug: 'pinterest' },
 	},
+	{
+		name: 'wolfram-cloud',
+		title: 'Wolfram Cloud',
+		icon: embedWolframIcon,
+		description: __( 'Embed Wolfram Cloud notebook content.' ),
+		patterns: [ /^https?:\/\/(www\.)?wolframcloud\.com\/obj\/.+/i ],
+		attributes: { providerNameSlug: 'wolfram-cloud' },
```
Comment (ntsekouras) [inline packages/block-library/src/embed/variations.js]: Should we add the `responsive: true` here?

**Review by ntsekouras — APPROVED** [review]
Thanks! 👍

File: packages/block-library/src/embed/variations.js (hunk 1 above)
Comment (pento) [reply packages/block-library/src/embed/variations.js]: 👍🏻

**Pull request merged**